### PR TITLE
Make HttpSocket handle HTTPS requests via an HTTP proxy

### PR DESCRIPTION
**Pull request by RichieB2B**
Simple fix for issue #2057

### CODE REVIEW

File: lib/Cake/Network/CakeSocket.php [hunk 1]
```diff
@@ -170,6 +170,25 @@ public function connect() {
 		if ($this->connected) {
 			stream_set_timeout($this->connection, $this->config['timeout']);
 		}
+
+		if (!empty($this->config['request']) && $this->config['request']['uri']['scheme'] == 'https' && !empty($this->config['proxy'])) {
```
Comment (dereuromark) [inline lib/Cake/Network/CakeSocket.php]: strict === please

File: lib/Cake/Network/CakeSocket.php [hunk 2]
```diff
@@ -170,6 +170,25 @@ public function connect() {
...
+			$req = array();
+			$req[] = 'CONNECT '. $this->config['request']['uri']['host'] . ':' . $this->config['request']['uri']['port'] . ' HTTP/1.1';
+			$req[] = 'Host: ' . $this->config['host'];
+			$req[] = 'User-Agent: php proxy';
+
+			fwrite($this->connection, implode("\r\n", $req)."\r\n\r\n");
+
+			while(true) {
+				$s = rtrim(fgets($this->connection, 4096));
+				if(preg_match('/^$/', $s)) {
```
Comment (dereuromark) [inline lib/Cake/Network/CakeSocket.php]: missing space after `if`

**Comment by RichieB2B** [comment]
Fixed the two comments by @dereuromark

File: lib/Cake/Network/CakeSocket.php [hunk 3]
```diff
@@ -170,6 +170,25 @@ public function connect() {
...
 			stream_set_timeout($this->connection, $this->config['timeout']);
 		}
+
+		if (!empty($this->config['request']) && $this->config['request']['uri']['scheme'] === 'https' && !empty($this->config['proxy'])) {
+			$req = array();
+			$req[] = 'CONNECT '. $this->config['request']['uri']['host'] . ':' . $this->config['request']['uri']['port'] . ' HTTP/1.1';
+			$req[] = 'Host: ' . $this->config['host'];
+			$req[] = 'User-Agent: php proxy';
+
+			fwrite($this->connection, implode("\r\n", $req)."\r\n\r\n");
```
Comment (markstory) [inline lib/Cake/Network/CakeSocket.php]: Need spaces around the `.`.

File: lib/Cake/Network/CakeSocket.php [hunk 4]
```diff
@@ -170,6 +170,25 @@ public function connect() {
...
+
+			while(true) {
```
Comment (markstory) [inline lib/Cake/Network/CakeSocket.php]: Should be `while (true) {`

File: lib/Cake/Network/CakeSocket.php [hunk 5]
```diff
@@ -170,6 +170,25 @@ public function connect() {
...
+				$s = rtrim(fgets($this->connection, 4096));
+				if (preg_match('/^$/', $s)) {
+					break;
+				}
```
Comment (markstory) [inline lib/Cake/Network/CakeSocket.php]: What happens if this case never happens. You're in an infinite loop. Instead should you be checking `while (feof($this->connection)) {`?

**Comment by RichieB2B** [comment]
Good point about using feof(). According to the documentation this should also take care of a timeout.

**Comment by RichieB2B** [comment]
This pull request has been replaced by #6146

**Pull request closed without merging**